packages/admin-next/dashboard/src/routes/orders/order-edit/components/order-edit-form/order-edit-form.tsx

```diff
@@ -0,0 +1,354 @@
+    orderEdit.id
+  )
+  const { mutateAsync: updateOrderEdit } = useAdminUpdateOrderEdit(orderEdit.id)
+
+  const onQuantityChangeComplete = async (itemId: string) => {
+    setIsLoading(true)
+
+    const quantity = form.getValues()[itemId]
+
+    // if (form.getValues()[itemId] === 0) {
+    //   await medusa.admin.orderEdits.removeLineItem(orderEdit.id, itemId)
+    // } else
+
+    if (quantity > 0) {
+      await medusa.admin.orderEdits.updateLineItem(orderEdit.id, itemId, {
+        quantity,
+      })
+    }
+    await queryClient.invalidateQueries(
+      adminOrderEditsKeys.detail(orderEdit.id)
+    )
+
+    setIsLoading(false)
+  }
+
+  const currentItems = useMemo(
+    () =>
+      orderEdit?.items
+        .sort((i1, i2) => i1.id.localeCompare(i2.id))
+        .filter((i) => i.original_item_id) || [],
+    [orderEdit]
+  )
+  const addedItems = useMemo(
+    () =>
+      orderEdit?.items
+        .sort((i1, i2) => i1.id.localeCompare(i2.id))
+        .filter((i) => !i.original_item_id) || [],
+    [orderEdit]
+  )
+
+  /**
+   * EFFECTS
+   */
+  useEffect(() => {
+    if (orderEdit) {
+      orderEdit?.items.forEach((i) => {
+        form.setValue(i.id, i.quantity)
+      })
+    }
+  }, [orderEdit?.items.length])
+
+  /**
+   * HANDLERS
+   */
+  const handleOpenChange = (open: boolean) => {
+    if (!open) {
+      setSearchParams(
+        {},
+        {
+          replace: true,
+        }
+      )
+    }
+
+    setOpen(open)
+  }
+
+  const onVariantsSelect = async (variantIds: string[]) => {
+    setIsLoading(true)
+    try {
+      await Promise.all(
+        variantIds.map((id) =>
+          addLineItemToOrderEdit({ variant_id: id, quantity: 1 })
+        )
+      )
+
+      await queryClient.invalidateQueries(adminOrderKeys.detail(order.id))
+    } finally {
+      setIsLoading(false)
+    }
+
+    setOpen(false)
+  }
+
+  const onItemRemove = async (itemId: string) => {
+    setIsLoading(true)
+
+    const change = orderEdit.changes.find(
+      (change) =>
+        change.line_item_id === itemId ||
+        change.original_line_item_id === itemId
+    )
+    try {
+      if (change) {
+        if (change.type === "item_add") {
+          await medusa.admin.orderEdits.deleteItemChange(
+            orderEdit.id,
+            change.id
+          )
+        }
+        if (change.type === "item_update") {
+          await medusa.admin.orderEdits.deleteItemChange(
+            orderEdit.id,
+            change.id
+          )
+          await medusa.admin.orderEdits.removeLineItem(orderEdit.id, itemId)
+        }
+      } else {
+        await medusa.admin.orderEdits.removeLineItem(orderEdit.id, itemId)
+      }
+
+      await queryClient.invalidateQueries(
+        adminOrderEditsKeys.detail(orderEdit.id)
+      )
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    setIsLoading(true)
```

Comment:
Think we need to split up the isLoading state for and show them the in the correct contexts.